### PR TITLE
Feat: Delete scheduled campaign

### LINKED ISSUE #23: Schedule campaigns
## Endpoint

New endpoint(s) to schedule campaigns

Upsert endpoint
```
PATCH /campaigns/{id}/schedule

Request body:
schedule_date=2021-06-01 22:00:00

Response:
{
  "message": "Campaign <name> successfully scheduled at <date>"
}
```

Delete endpoint
```
DELETE /campaigns/{id}/schedule

Response:
{
  "message": "Campaign schedule removed successfully"
}
```

New table:

| id VARBINARY(27) PK | campaign_id FK(campaigns) |    scheduled_at     |     created_at      |     updated_at      |
|---------------------|---------------------------|---------------------|---------------------|---------------------|
| abcdef              |                       123 | 2021-06-01 20:00:00 | 2021-05-01 15:00:00 | 2021-05-01 15:00:00 |

* For each campaign we can have **only** one record
* On campaign delete, cascade delete scheduled campaign record
* Boundary checks need to be performed to see if the user has access to this feature
* The `scheduled_at` should not be before `now`
* The campaign should have `draft` status

## Start campaign changes

* If the user decides to start the campaign while it is scheduled, we should fetch the id from the schedule and send that as the `event_id`
* Update the campaign's status to `sending` and set the new column `event_id` from the scheduled record OR generate a new one in case the user has not scheduled the campaign beforehand

## Campaigner changes

* Remove the status check as an idempotency measure
* Use the same method as the sender consumer to handle idempotency
* The `send_logs` record should contain this new `event_id` 

## Sender changes

* The idempotency key should be a combination of the `event_id` and `subscriber_id`
* The `send_logs` record should contain this new `event_id`

## Campaign scheduler

We will have a cron scheduler that will fetch "expired" campaigns (scheduled_at < now) and publish message to the campaigner topic for all `draft` campaigns. We will set the `event_id` from the scheduled record's id

### PR DESCRIPTION
**Pull request by Dzalevski**
# Pull Request Template

## Description

API for deleting scheduled campaign

Fixes #23 

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A Unit test for storage/DeleteScheduledCampaign
- [ ] Test B Incoming Integration API test.
 
## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: actions/campaigns.go [hunk 1]
```diff
@@ -575,3 +575,41 @@ func GetCampaignBounces(c *gin.Context) {
...
+		return
+	}
+	campaign, err := storage.GetCampaign(c, id, user.ID)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "Campaign not found.",
+		})
+		return
+	}
+	if campaign.ID != id {
```
Comment (gudgl) [inline actions/campaigns.go:596]: Do we really need to check this id?

Comment (Dzalevski) [reply actions/campaigns.go:596]: @gud\ Yes because we are deleting only with campaign_id (user_id is not included in scheduled_campaign)

Comment (Dzalevski) [reply actions/campaigns.go:596]: yes you are right we do not need it.

Comment (gudgl) [reply actions/campaigns.go:596]: @Dzalevski you are fetching the campaign with this id why we need to check it again

File: actions/campaigns.go [hunk 2]
```diff
@@ -575,3 +576,44 @@ func GetCampaignBounces(c *gin.Context) {
...
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "Id must be an integer.",
+		})
+		return
+	}
+	campaign, err := storage.GetCampaign(c, id, u.ID)
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			c.JSON(http.StatusBadRequest, gin.H{
+				"message": "This campaign does not belong to you.",
```
Comment (FilipNikolovski) [inline actions/campaigns.go:594]: The error message should be the same as the one below, we don't know that this campaign actually exist or not, and the user will be misguided by this message.

**Review by FilipNikolovski — CHANGES_REQUESTED** [review]

File: actions/campaigns.go (hunk 2 above)
Comment (Dzalevski) [reply actions/campaigns.go:594]: @FilipNikolovski Then we don't need this checkup

**Review by FilipNikolovski — APPROVED** [review]

**Pull request merged**